libs/sparkpost-lib/src/main/java/com/sparkpost/transport/IRestConnection.java

```diff
@@ -0,0 +1,67 @@
+
+package com.sparkpost.transport;
+
+import com.sparkpost.exception.SparkPostException;
+import com.sparkpost.model.responses.Response;
+
+public interface IRestConnection {
+
+    /**
+     * Default endpoint to use for connections :
+     * https://api.sparkpost.com/api/v1
+     */
+    String defaultApiEndpoint = "https://api.sparkpost.com/api/v1";
+
+    /**
+     * Perform an HTTP GET request. This method throws an exception if the
+     * server returns anything else than a 200.
+     *
+     * @param path
+     *            API endpoint to send the request to.
+     * @return Server response to the request.
+     * @throws SparkPostException
+     *             if something goes wrong
+     */
+    Response get(String path) throws SparkPostException;
```

Comment:
Should the `IRestConnection` methods accept an `Endpoint` rather than a `String`? That would ensure use of Endpoint and let it and the `RestConnection` impl enforce our single slash requirement. It'd presumably simplify client code a little too.

That's a breaking change though so maybe adding Endpoint versions of the methods and `@deprecating` the old String versions would be the way to go.